commands/summary.go

```diff
@@ -274,6 +235,103 @@ func (cmd *summaryCommand) makeArchive() (err error) {
 	return //nolint:nakedret
 }
 
+type profilerPath struct {
+	suffix  string
+	webPath string
+}
+
+func (cmd *summaryCommand) getPprofData() []string {
+	profilerPaths := []profilerPath{
+		{
+			suffix:  "profile.pb.gz",
+			webPath: "/profile?seconds=60",
+		},
+		{
+			suffix:  "heap.pb.gz",
+			webPath: "/heap?gc=1",
+		},
+		{
+			suffix:  "trace.out",
+			webPath: "/trace?seconds=10",
+		},
+	}
+	apps := map[string]string{
+		"pmm-agent": "http://127.0.0.1:7777/debug/pprof",
+	}
+	if !cmd.SkipServer {
+		apps["pmm-managed"] = "http://127.0.0.1:7773/debug/pprof"
+		apps["qan-api2"] = "http://127.0.0.1:9933/debug/pprof"
+	}
+
+	out := make(chan string)
+	files := make([]string, 0)
+	wg := &sync.WaitGroup{}
+	wgr := &sync.WaitGroup{}
+
+	wgr.Add(1)
+
+	go func() {
+		for fileName := range out {
+			files = append(files, fileName)
+		}
+
+		wgr.Done()
+	}()
+
+	for appName, baseUrl := range apps {
+		wg.Add(1)
+		go func(appName, baseUrl string) {
+			defer wg.Done()
+
+			for _, file := range profilerPaths {
+				fs := fmt.Sprintf("%s-%s", appName, file.suffix)
+				url := baseUrl + file.webPath
+
+				downloadProfilerData(url, fs, out)
+			}
+		}(appName, baseUrl)
+	}
+
+	wg.Wait()
+	close(out)
+
+	wgr.Wait()
+	return files
+}
+
+func downloadProfilerData(url string, fs string, out chan string) {
+	logrus.Debugf("Started downloading profiler data from %s", url)
+	resp, err := http.Get(url) //nolint
+	if err != nil {
+		logrus.Errorf("cannot get profiles info: %s", err)
+		return
+	}
+	if resp.StatusCode != http.StatusOK {
+		err := fmt.Errorf("cannot get profiler data(%s). Status code: %d", url, resp.StatusCode)
+		logrus.Errorf("cannot get profiles info: %s", err)
+		return
+	}
+
+	tmpfile, err := ioutil.TempFile("", "*_"+fs)
+	if err != nil {
+		logrus.Errorf("cannot get profiles info: %s", "cannot create temp file")
+		return
+	}
+
+	if _, err := io.Copy(tmpfile, resp.Body); err != nil {
+		logrus.Errorf("cannot get profiles info: %s", "cannot write pprof file")
```

Comment:
We don't close a temporary file in that branch. Go runtime uses a finalizer for file descriptors to try to mitigate the problem of them leaking, but it is not bullet-proof.

In that case, seems like `defer tmpfile.Close()` is enough.

But then we are also not removing temporary files? Then it is probably better to modify zip functions to accept `[]byte`.